kubernetes/kubernetes_test.go

```diff
@@ -519,5 +621,62 @@ func getMockInformers() map[string]*cache.SharedIndexInformer {
 		"jobs":                   &jinformer,
 		"cronjobs":               &cjinformer,
 	}
+	// Call the Run function for each Informer, allowing the informers to listen for Add events
+	startMockInformers(mockInformers)
+
+	// Private Annotation to be removed if ParseMetrics is enabled
+	annotation := map[string]string{
+		"kubectl.kubernetes.io/last-applied-configuration": "IAmSecretEnvVariables",
+	}
+
+	// now add items to informers after they are running
+	replicationControllers.Add(&v1.ReplicationController{ObjectMeta: metav1.ObjectMeta{Name: "rc1", Annotations: annotation}})
+	services.Add(&v1.Service{ObjectMeta: metav1.ObjectMeta{Name: "s1", Annotations: annotation}})
+	nodes.Add(&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "n1", Annotations: annotation}})
+	persistentVolumes.Add(&v1.PersistentVolume{ObjectMeta: metav1.ObjectMeta{Name: "pv1", Annotations: annotation}})
+	persistentVolumeClaims.Add(&v1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{Name: "pvc1", Annotations: annotation}})
+	replicaSets.Add(&v1apps.ReplicaSet{ObjectMeta: metav1.ObjectMeta{Name: "rs1", Annotations: annotation}})
+	daemonSets.Add(&v1apps.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "ds1", Annotations: annotation}})
+	jobs.Add(&v1batch.Job{ObjectMeta: metav1.ObjectMeta{Name: "job1", Annotations: annotation}})
+	if clusterVersion > 1.20 {
+		cronJobs.Add(&v1batch.CronJob{ObjectMeta: metav1.ObjectMeta{Name: "cj1", Annotations: annotation}})
+	}
+
+	// pods is unique as we use this pod file for parseMetrics testing
+	// for parseMetricData testing, add a cldy metrics-agent pod to the mock informers
+	podData, _ := ioutil.ReadFile("../testdata/pods.jsonl")
+	var myPod *v1.Pod
+	err := json.Unmarshal(podData, &myPod)
+	if err != nil {
+		t.Error(err)
+	}
+	pods.Add(myPod)
+	// namespace also used in testing
+	namespaceData, _ := ioutil.ReadFile("../testdata/namespaces.jsonl")
+	var myNamespace *v1.Namespace
+	err = json.Unmarshal(namespaceData, &myNamespace)
+	if err != nil {
+		t.Error(err)
```

Comment:
Ideally should return an error in the cases and let the caller handle it.

Reply:
Sounds good, refactored